xbmc/cores/RetroPlayer/buffers/RenderBufferDMA.cpp

```diff
@@ -50,11 +50,13 @@ size_t CRenderBufferDMA::GetFrameSize() const
 
 uint8_t* CRenderBufferDMA::GetMemory()
 {
+  m_bo->SyncStart();
   return m_bo->GetMemory();
 }
 
 void CRenderBufferDMA::ReleaseMemory()
 {
+  m_bo->SyncEnd();
```

Comment:
If the documentation is `Must be called after reading/writing data to the BufferObject` then shouldn't these two lines be swapped?

Reply:
I don't really think mapping/unmapping matters as we should already be dont writting at this point.
```
 - mmap dma-buf fd
 - for each drawing/upload cycle in CPU 1. SYNC_START ioctl, 2. read/write to mmap area 3.
 SYNC_END ioctl. This can be repeated as often as you want (with the new data being consumed by say the GPU or the scanout device)
 - munmap once you don't need the buffer any more
```

Reply:
It looks like these lines should be swapped. read/write doesn't finish until release, and 2. needs to finish before 3.

Reply:
updated, thanks for the input